schema/docs/docs.go

```diff
@@ -95,22 +96,26 @@ func makeParserMapForPackage(pkg string) (map[string]*ast.Package, error) {
 		if _, ok := parserMapCache[current]; ok {
 			continue
 		}
-		if !strings.HasPrefix(current, root) &&
-			!strings.HasPrefix(current, ".") &&
-			!strings.HasPrefix(current, "vendor/") { // TODO: this will not work with go mod cache as written.
-			current = "vendor/" + current
+		pkgDir := current
+		if !strings.HasPrefix(pkgDir, root) && !strings.HasPrefix(pkgDir, ".") {
+			vendoredPkg := path.Join("vendor", pkgDir)
+			if !strings.HasPrefix(pkgDir, "vendor") && dirExists(vendoredPkg) {
+				// vendor mode
+				pkgDir = vendoredPkg
+			} else {
+				pkgDir = path.Join(gomodcache(), pkgDir)
```

Comment:
This is not correct.

We need to join: `GOMODCACHE`, dep.name+`@`dep.version, rest-of-the-package (see [similar code](https://github.com/openshift-knative/hack/pull/585/files#diff-70f4fd225b195b7c6a8427350815e9f81f965e2627933ed3f9d3d37e2a42d885R97))

To do it we need the current gomod file

Reply:
/hold